Source/Core/FeatureDetection.js

```diff
@@ -199,6 +203,37 @@ define([
         return supportsImageRenderingPixelated() ? imageRenderingValueResult : undefined;
     }
 
+    var supportsWebpPromise;
+    function supportsWebp() {
+        // From https://developers.google.com/speed/webp/faq#how_can_i_detect_browser_support_for_webp
+        if (defined(supportsWebpPromise)) {
+            return supportsWebpPromise;
```

Comment:
This should be `supportsWebpPromise.promise`, you are currently returning a `deferred`, not a promise (which is different in subtle ways).